azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -149,6 +149,8 @@ def _setup_permanent_firewalld_rules(self):
             return
 
         logger.info("Firewall rules not added yet, adding them now using firewalld.service")
+        # Remove first if partial list present
+        AddFirewallRules.remove_firewalld_rules(self._dst_ip, self._uid)
```

Comment:
Thanks for adding this here, make sure to add a test for this too in DCR (along with the normal iptable tests)